src/mscorlib/src/System/Text/StringBuilder.cs

```diff
@@ -281,24 +281,30 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             }
         }
 
-        public int Capacity {
+        public int Capacity
+        {
             get { return m_ChunkChars.Length + m_ChunkOffset; }
-            set {
-                if (value < 0) {
+            set
+            {
+                if (value < 0)
+                {
                     throw new ArgumentOutOfRangeException("value", Environment.GetResourceString("ArgumentOutOfRange_NegativeCapacity"));
                 }
-                if (value > MaxCapacity) {
+                if (value > MaxCapacity)
+                {
                     throw new ArgumentOutOfRangeException("value", Environment.GetResourceString("ArgumentOutOfRange_Capacity"));
                 }
-                if (value < Length) {
+                if (value < Length)
+                {
                     throw new ArgumentOutOfRangeException("value", Environment.GetResourceString("ArgumentOutOfRange_SmallCapacity"));
                 }
                 Contract.EndContractBlock();
 
-                if (Capacity != value) {
+                if (Capacity != value)
+                {
                     int newLen = value - m_ChunkOffset;
                     char[] newArray = new char[newLen];
-                    Array.Copy(m_ChunkChars, newArray, m_ChunkLength);
+                    Buffer.BlockCopy(m_ChunkChars, 0, newArray, 0, m_ChunkLength * sizeof(char));
```

Comment:
All code should be written such that it works fine with 2GB+ arrays, independent on whether 2GB arrays are supported on CoreCLR array. (I think you can enable them via env variable, but it is not officially documented.)

Once you do all this, the change will become pretty complex - for a very limited benefit.

What I would be ok with here is to use `Array.Copy` overload that takes explicitly indices - it would save the GetLowerBound calls (that always return zero). But I do not think it is worth the complexity to change this code to use a different memory copy method.